pkg/corerp/frontend/controller/applications/graph_util.go

```diff
@@ -0,0 +1,581 @@
+	sort.Slice(entries, func(i, j int) bool {
+		// //if entries[i].Provider != entries[j].Provider {
+		// 	return entries[i].Provider < entries[j].Provider
+		// }
+		if entries[i].Type != entries[j].Type {
+			return *entries[i].Type < *entries[j].Type
+		}
+		if entries[i].Name != entries[j].Name {
+			return *entries[i].Name < *entries[j].Name
+		}
+		return *entries[i].ID < *entries[j].ID
+
+	})
+
+	return entries
+}
+
+// connectionsFromAPIData resolves the outbound connections for a resource from the generic resource representation.
+// For example if the resource is an 'Applications.Core/container' then this function can find it's connections
+// to other resources like databases. Conversely if the resource is a database then this function
+// will not find any connections (because they are inbound). Inbound connections are processed later.
+func connectionsFromAPIData(resource generated.GenericResource) []*corerpv20231001preview.ApplicationGraphConnection {
+	// We need to access the connections in a weakly-typed way since the data type we're
+	// working with is a property bag.
+	//
+	// Any Radius resource type that supports connections uses the following property path to return them.
+	p, err := jsonpointer.New("/properties/connections")
+	if err != nil {
+		// This should never fail since we're hard-coding the path.
+		panic("parsing JSON pointer should not fail: " + err.Error())
+	}
+
+	raw, _, err := p.Get(&resource)
+	if err != nil {
+		// Not found, this is fine.
+		return []*corerpv20231001preview.ApplicationGraphConnection{}
+	}
+
+	connections, ok := raw.(map[string]any)
+	if !ok {
+		// Not a map of objects, this is fine.
+		return []*corerpv20231001preview.ApplicationGraphConnection{}
+	}
+
+	// The data is returned as a map of JSON objects. We need to convert each object from a map[string]any
+	// to the strongly-typed format we understand.
+	//
+	// If we encounter an error processing this data, just skip "invalid" connection entry.
+	entries := []*corerpv20231001preview.ApplicationGraphConnection{}
+	for _, connection := range connections {
+		dir := corerpv20231001preview.DirectionInbound
+		data := corerpv20231001preview.ConnectionProperties{}
+		err := toStronglyTypedData(connection, &data)
+		if err == nil {
+			entries = append(entries, &corerpv20231001preview.ApplicationGraphConnection{
+				ID:        data.Source,
+				Direction: &dir,
+			})
+		}
+	}
+
+	// Produce a stable output
+	sort.Slice(entries, func(i, j int) bool {
+		return *entries[i].ID < *entries[j].ID
+	})
+
+	return entries
+}
+
+func providesFromAPIData(resource generated.GenericResource) []*corerpv20231001preview.ApplicationGraphConnection {
+	// We need to access the connections in a weakly-typed way since the data type we're
+	// working with is a property bag.
+	//
+	// Any Radius resource type that supports connections uses the following property path to return them.
+	p, err := jsonpointer.New("/properties/container/ports")
```

Comment:
Why are we checking this path?

Reply:
This is to specifically suport httproutes. Since ports has the provides , which is needed to be parsed to understand the connections formed by httproutes. container A provides rte. cntainer B consumes route . Translates to "port--> provides "section in container A and "connection" construct in container B thus we will get container A--> rte --> container B. else container A--> rte will be missing from the graph.

Reply:
Thanks for explaining. Could you please add this in the comments?

Reply:
added